plugins/shrs_completion/src/lib.rs

```diff
@@ -4,3 +4,65 @@
 
 pub mod completions;
 mod helpers;
+use std::{fs, rc::Rc};
+
+use ::anyhow::anyhow;
+use dirs::home_dir;
+use rhai::{Array, Engine, Scope};
+use setup::setup_engine;
+use shrs::prelude::*;
+
+pub struct CompletionsPlugin;
+pub mod setup;
+
+impl Plugin for CompletionsPlugin {
+    fn init(&self, _: &mut ShellConfig) -> anyhow::Result<()> {
+        Ok(())
+    }
+    fn post_init(&self, sh: &Shell, ctx: &mut Context, rt: &mut Runtime) -> ::anyhow::Result<()> {
+        let mut e = Engine::new();
+        setup_engine(&mut e);
+        let engine = Rc::new(e);
+        let mut folder = home_dir().unwrap();
+        folder.push(".config/shrs/completions");
+        for p in fs::read_dir(folder).unwrap() {
+            let path = p.unwrap().path();
+            let compiled = engine.compile_file(path);
+            let ast = Rc::new(match compiled {
+                Ok(ast) => ast,
+                Err(e) => {
+                    eprintln!("Rhai script compile error {}", e);
+                    return Err(anyhow!("Can't compile"));
+                },
+            });
+            let ast1 = ast.clone();
+
+            let e1 = engine.clone();
+            let e2 = engine.clone();
+
+            ctx.completer.register(Rule::new(
+                Pred::new(move |c| {
+                    let mut scope = Scope::new();
+
+                    let predicate: bool = e1
+                        .call_fn::<bool>(&mut scope, &ast, "predicate", (c.clone(),))
+                        .unwrap();
```

Comment:
display user warning if they have malformed rhai completion script; since this is user supplied code we should be a bit more careful about unwraps

Reply:
maybe we can just skip the completion if something goes wrong and print a warning rather than aborting the entire shell